### DISCUSSION TITLE
Rename or remove addMarkdownHighlighter

### DISCUSSION THREAD

**Discussion by Snapstromegon** [11ty/eleventy#4145] [Enhancement Queue]
### Is your feature request related to a problem? Please describe.

In general the `add*()` functions of the eleventy config are non-destructive if possible. `addMarkdownHightlighter()` on the other hand behaves like the `set*()` functions.

Also this function IMO is no longer necessary, since there are other, better and better documented options in eleventy.

Function in code:
https://github.com/11ty/eleventy/blob/main/src/UserConfig.js#L170

### Describe the solution you'd like

I would recommend to either remove this function completely with the next major version, or to rename it to `setMarkdownHighlighter()` and add documentation for it.

### Describe alternatives you've considered

Not changing anything is possible, but leaves this in a weird state.

**Comment by Subham-KRLX** [comment]
Hi! I’d like to work on this issue. May I take it up?

**Comment by mohammedadelabu** [comment]
Hi! I’d like to work on this issue. Can I pick it up?

**Comment by mohammed-hassan-angular-frontend-fcmb** [comment]
Hi! I would like to work on this issue. May I take it up?

**Comment by zachleat** [comment]
Hmm, I wouldn’t consider this part of the public userland API but it does exist and we’ll probably just leave it as-is for backwards compatibility.

For competeness, it is used by the syntax highlighter plugin v5 and will be removed by v6 https://github.com/11ty/eleventy-plugin-syntaxhighlight/pull/100 so we’ll probably wait awhile after v6 is released to discuss removing it from core at that point
